### PR TITLE
Utilize i.n.u.internal.ObjectUtil Preconditions (handler) (#11170)

### LINKED ISSUE #11170: Utilize i.n.u.internal.ObjectUtil to assert Preconditions
### Expected behavior
NullChecks resulting in a NullPointerException or IllegalArgumentException, numeric ranges (>0, >=0) checks, not empty strings/arrays checks must never be anonymous but with the parameter or variable name which is checked. They must be specific and should not be done with an "OR-Logic" (if a == null || b == null) throw new NullPointerEx.

### Actual behavior
Some checks are already done with the io.netty.util.internal.ObjectUtil code. Some are done manually.

### Steps to reproduce
checkout the master
grep -r "Objects.req"
grep -r "throw new NullPointerException"
grep -r "throw new IllegalArgumentException" (individual check necessary)

### Minimal yet complete reproducer code (or URL to code)
Existing codebase.

### Netty version
4.1 (updated version)

### JVM version (e.g. `java -version`)
Not relevant, all
### OS version (e.g. `uname -a`)
Not relevant, all

### Providing the fix
One pull request per sub-project to enable reviewers to focus on a relevant subset. First one is common itself, including a new unit test for ObjectUtil itself.

### PR DESCRIPTION
**Pull request by boris-unckel**
Motivation:

NullChecks resulting in a NullPointerException or IllegalArgumentException, numeric ranges (>0, >=0) checks, not empty strings/arrays checks must never be anonymous but with the parameter or variable name which is checked. They must be specific and should not be done with an "OR-Logic" (if a == null || b == null) throw new NullPointerEx.

Modifications:

* import static relevant checks
* Replace manual checks with ObjectUtil methods

Result:

All checks needed are done with ObjectUtil, some exception texts are improved.

Fixes #11170
One PR for each subproject to ease review.

### CODE REVIEW

File: handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java [hunk 1]
```diff
@@ -1090,16 +1087,14 @@ public final SSLEngineResult unwrap(
             final ByteBuffer[] dsts, int dstsOffset, final int dstsLength) throws SSLException {
 
         // Throw required runtime exceptions
-        ObjectUtil.checkNotNull(srcs, "srcs");
+        checkNotNullWithIAE(srcs, "srcs");
```
Comment (normanmaurer) [inline handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java:1090]: While this is different to what happened before (before it was a NPE) I think the new behaviour is more correct

Comment (boris-unckel) [reply handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java:1090]: I should have commented this one. I compared it to the other checks and switched it therefor.

File: handler/src/main/java/io/netty/handler/traffic/GlobalChannelTrafficShapingHandler.java [hunk 2]
```diff
@@ -299,15 +300,13 @@ public void setMaxDeviation(float maxDeviation, float slowDownFactor, float acce
...
-            throw new IllegalArgumentException("slowDownFactor must be >= 0");
-        }
+
         if (accelerationFactor > 0) {
             throw new IllegalArgumentException("accelerationFactor must be <= 0");
         }
         this.maxDeviation = maxDeviation;
         this.accelerationFactor = 1 + accelerationFactor;
-        this.slowDownFactor = 1 + slowDownFactor;
+        this.slowDownFactor = 1 + checkPositive(slowDownFactor, "slowDownFactor");
```
Comment (normanmaurer) [inline handler/src/main/java/io/netty/handler/traffic/GlobalChannelTrafficShapingHandler.java:309]: this change makes the behaviour different to what we have now.. as you moved the positive check here we may already upped some fields before throwing. Please move the check before assignments

**Review by normanmaurer — CHANGES_REQUESTED** [review]

File: handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java (hunk 1 above)
Comment (normanmaurer) [reply handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java:1090]: no problem... like I said I think it is now more correct :)

File: handler/src/main/java/io/netty/handler/traffic/GlobalChannelTrafficShapingHandler.java [hunk 3]
```diff
@@ -299,9 +300,7 @@ public void setMaxDeviation(float maxDeviation, float slowDownFactor, float acce
         if (maxDeviation > MAX_DEVIATION) {
             throw new IllegalArgumentException("maxDeviation must be <= " + MAX_DEVIATION);
         }
-        if (slowDownFactor < 0) {
-            throw new IllegalArgumentException("slowDownFactor must be >= 0");
-        }
+        checkPositive(slowDownFactor, "slowDownFactor");
         if (accelerationFactor > 0) {
```
Comment (normanmaurer) [inline handler/src/main/java/io/netty/handler/traffic/GlobalChannelTrafficShapingHandler.java:304]: @boris-unckel also fix this one ?

**Review by normanmaurer — CHANGES_REQUESTED** [review]

File: handler/src/main/java/io/netty/handler/traffic/GlobalChannelTrafficShapingHandler.java (hunk 3 above)
Comment (boris-unckel) [reply handler/src/main/java/io/netty/handler/traffic/GlobalChannelTrafficShapingHandler.java:304]: Sorry, yes, sure.

**Review by normanmaurer — APPROVED** [review]

**Pull request merged**